### ISSUE TITLE
Runfile dependencies for go_proto_compiler 

### ISSUE THREAD

**Issue by fischor** [bazelbuild/rules_go#2704]
### What version of rules_go are you using?

v0.24.5

### What version of gazelle are you using?

v0.22.2

### What version of Bazel are you using?

3.7.0

### What did you do?

I was trying to build a protoc plugin that uses a configuration file, ideally passed using the options in the protoc command. 
I think there is no way to achieve this atm.  

There is no way to add a `data` to the `go_proto_compiler` rule directly (and then access it with the plugin). 

Thus, I wanted to pass the config file as a runfile to the plugin target like this:

```python
    go_binary(
        name = "protoc-gen-validate",
        embed = [":protoc-gen-validate_lib"],
        data = [":my-config-file.yaml"],
    )
    go_proto_compiler(
        name = "my-compiler",
        options = options,
        plugin = ":protoc-gen-validate",
        suffix = ".validate.pb.go",
    )
```

But also this did not work, because when running the `go_proto_library` target, the `go_binary` had no access to the provided `data` runfiles anymore. 
I think thats because the `go_proto_library` handles the plugin as a file and not as an executable? Would the runfiles be accessible if it would be treated as an executable?

Is there a way to achieve what I want?

So in summary, this is a feature request to either have some kind of `data` attribute on the `go_proto_compiler` and make this data available for the plugin when run, or to treat the `plugin` as an executable. 

 Would you accept a PR for this?

**Comment by bshi** [comment]
+1; One concrete use case would be to support a protoc plugin option [`--grpc-gateway_opt grpc_api_configuration=...`](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/docs/docs/grpcapiconfiguration.md#usage-of-grpc-api-configuration-yaml-files)

**Comment by jayconrod** [comment]
Marking as a bug. It doesn't look like the `go_proto_compile` function in `proto/compile.bzl` does anything with runfiles when it creates the action. This might be as simple as passing in the plugin as a tool instead of an input.
